src/app/features/profile/interfaces/profile-state.ts

```diff
@@ -1,5 +1,7 @@
 import { UserProfile } from './user-profile';
+import { UserProfileMap } from './user-profile-map';
 
 export interface ProfileState {
     user?: UserProfile;
+    usersMap: UserProfileMap;
```

Comment:
We could eliminate the `user` prop and instead add the fetched profile to the map when the route doesn't contain a `profileId`.  This allows for a more normalized store when selecting from the list, but I think that would imply the need to update the URL with the id once fetched, so I've not included that here.